### PR TITLE
Small modifications for null-safety

### PR DESCRIPTION
**Pull request by srawlins**
These include nullability hints for the migration tool, local variables for
non-promotable fields, and local variables for regex matches.

### CODE REVIEW

File: tool/stats_lib.dart [hunk 1]
```diff
@@ -74,10 +74,10 @@ class CommonMarkTestCase {
   factory CommonMarkTestCase.fromJson(Map<String, dynamic> json) {
     return CommonMarkTestCase(
         json['example'] as int,
-        json['section'] as String,
+        json['section'] as String /*!*/,
```
Comment (kevmoo) [inline tool/stats_lib.dart:77]: This (and below) seem really weird. `as String` should completely imply "not null". `!` shouldn't me needed!

Comment (kevmoo) [reply tool/stats_lib.dart:77]: I guess when migration is done these will be eliminated. This is just to hint the tool?

**Comment by kevmoo** [comment]
Are you going to create a `null-safety` branch here soonish?

File: tool/stats_lib.dart (hunk 1 above)
Comment (srawlins) [reply tool/stats_lib.dart:77]: Yeah, the migration tool sometimes decides that that should be a nullable string.

**Comment by srawlins** [comment]
Haha I don't desire to. markdown only depends on args and charcode though, so I certainly could. It might be after 2.9.0 final is released though, which would be nice, as I could avoid the branch then.

**Comment by kevmoo** [comment]
> Haha I don't desire to. markdown only depends on args and charcode though, so I certainly could. It might be after 2.9.0 final is released though, which would be nice, as I could avoid the branch then.

`args` is going to be "fun" 😄

**Pull request merged**